debezium-embedded/src/main/java/io/debezium/embedded/ConvertingEngineBuilder.java

```diff
@@ -165,10 +166,18 @@ public void markProcessed(R record, DebeziumEngine.Offsets sourceOffsets) throws
             toFormat = (record) -> {
                 final byte[] key = keyConverter.fromConnectData(TOPIC_NAME, record.keySchema(), record.key());
                 final byte[] value = valueConverter.fromConnectData(TOPIC_NAME, record.valueSchema(), record.value());
-                return (R) new EmbeddedEngineChangeEvent<String, String>(
-                        key != null ? new String(key) : null,
-                        value != null ? new String(value) : null,
-                        record);
+                String formatJson = Json.class.getSimpleName();
+                String formatCloudEvents = CloudEvents.class.getSimpleName();
+                return formatKey.getSimpleName().equals(formatJson) && formatValue.getSimpleName().equals(formatJson)
```

Comment:
Do you need to compare the class names? Can't you reuse `io.debezium.embedded.ConvertingEngineBuilder.isFormat` ?